src/async_api/mod.rs

```diff
@@ -48,6 +48,9 @@ mod opened_session;
 mod reply;
 mod request;
```

Comment:
An audit is a good idea and i do like the separation -- but my only concern/hesitation here is that we would be flattening the mod tree as seen by users. Since we have multiple implemenations they would all be flat at the root with the re-export right? 

Its something akin to we need to remove everythiong from async_api and have it just be a re-export of an interior real mod maybe? Or do you think its nicer flattened to the lib.rs at the root?

Reply:
(We have multiple Reply types for instance for async vs sync now)

Reply:
Ah yes, that makes sense about having them in a separate module, and not flattened. I think you can still do that in `lib.rs` like:
```
pub mod async_api {
   pub use async_api::Filesystem;
}
```
Although the module names might need to be changed to not conflict.

The thing that I care about is that we don't export an entire module or wildcard, because things may leak into the public API by accident then.

Reply:
👍  i think is a great goal/plan - and I’ll do that, if we need to rename the api then that seems fine too

Reply:
Done, it required adding another module layer in... but it works